frontend/src/pages/generators/LetterGenerator.tsx

```
@@ -0,0 +1,471 @@
+        <div className="fixed inset-0 z-[100] flex items-center justify-center p-4 bg-background/90 backdrop-blur-md">
+          <div className="bg-card border border-border w-full max-w-xl rounded-[2.5rem] shadow-2xl p-10 space-y-8">
+            <div className="flex justify-between items-center border-b border-border pb-6">
+              <h3 className="text-2xl font-black text-foreground tracking-tighter uppercase">{previewTemplate.name}</h3>
+              <div className="px-3 py-1 bg-primary/10 text-primary text-[10px] font-black rounded-full uppercase tracking-widest">
+                {previewTemplate.type}
+              </div>
+            </div>
+
+            <div className="space-y-4">
+              <label className="text-[10px] font-black text-muted-foreground tracking-widest uppercase opacity-70">Letter Structure</label>
+              <div className="p-8 bg-muted/30 rounded-[2rem] border-2 border-dashed border-border/50 text-foreground font-bold text-lg text-center flex flex-col gap-4">
```

Biome (2.3.13)

[error] A form label must be associated with an input.

Consider adding a `for` or `htmlFor` attribute to the label element or moving the input element to inside the label element.

(lint/a11y/noLabelWithoutControl)